dynamiqs/solvers/result.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from torch import Tensor
+
+from ..options import Options
+
+
+def memory_bytes(x: Tensor) -> int:
+    return x.element_size() * x.numel()
+
+
+def memory_str(x: Tensor) -> str:
+    mem = memory_bytes(x)
+    if mem < 1024**2:
+        return f'{mem / 1024:.2f} Kb'
+    elif mem < 1024**3:
+        return f'{mem / 1024**2:.2f} Mb'
+
+    return f'{mem / 1024**3:.2f} Gb'
```

Comment:
```suggestion
    else:
        return f'{mem / 1024**3:.2f} Gb'
```
Personal preference, a bit more readable :D